uniflow/op/model/neuron_utils.py

```diff
@@ -0,0 +1,264 @@
+import hashlib
+import os
+import zipfile
+
+import requests
+import torch
+from tqdm import tqdm
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+try:
+    from transformers_neuronx import constants
+    from transformers_neuronx.config import NeuronConfig
+    from transformers_neuronx.generation_utils import HuggingFaceGenerationModelAdapter
+    from transformers_neuronx.mistral.model import MistralForSampling
+    from transformers_neuronx.module import save_pretrained_split
+except ModuleNotFoundError as exc:
+    raise ModuleNotFoundError(
+        "Please install transformers_neuronx. You can use `pip install transformers-neuronx --extra-index-url=https://pip.repos.neuron.amazonaws.com` to install it."
+    ) from exc
+
+
+class Neuron:
+    """
+    Neuron class.
+    """
+
+    neuron_model_config = {
+        "mistralai/Mistral-7B-Instruct-v0.2": {
+            1: {
+                8192: [
+                    "https://cambioml-neuron-cache.s3.us-west-2.amazonaws.com/transformers_neuronx/Mistral-7B-Instruct-v0.2_tp2-bz1-8192.zip",
+                    "367aa7db73c1f82b5add0ad5c97d7a77",
+                ]
+            },
+            2: {
+                4096: [
+                    "https://cambioml-neuron-cache.s3.us-west-2.amazonaws.com/transformers_neuronx/Mistral-7B-Instruct-v0.2_tp2-bz2-4096.zip",
+                    "f2f07a6d6b05e39b7e9d94cd6c95fb96",
+                ]
+            },
+            4: {
+                2048: [
+                    "https://cambioml-neuron-cache.s3.us-west-2.amazonaws.com/transformers_neuronx/Mistral-7B-Instruct-v0.2_tp2-bz4-2048.zip",
+                    "1c62a06d6f25c2241b72b95e00ededfb",
+                ]
+            },
+        }
+    }
+    instance_type_dict = {
+        "inf2.xlarge": 2,
+        "inf2.8xlarge": 2,
+        "inf2.24xlarge": 12,
+        "inf2.48xlarge": 24,
+    }
+
+    @staticmethod
+    def get_instance_type():
+        """
+        Get the instance type of the current instance.
+        """
+        url = "http://169.254.169.254/latest/meta-data/instance-type"
```

Comment:
nit: can you link https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html as a comment here for readability.